src/rust.rs

```diff
@@ -320,6 +321,50 @@ unsafe impl Trace for Heap<jsid> {
     }
 }
 
+pub trait AsHandle<T> {
+    fn handle(&self) -> Handle<T>;
+
+    fn get(&self) -> T where T: Copy {
+        self.handle().get()
+    }
+}
+
+/// Given the types `$T` and `$U`, implements the trait `AsHandle<$T>` for the type `$U`, where
+/// `$U` is a newtype whose underlying type implements `AsHandle<$T>`.
+#[macro_export]
+macro_rules! derive_as_handle {
+    ($T:ty, $U:ty) => {
+        impl $crate::rust::AsHandle<$T> for $U {
+            fn handle(&self) -> $crate::jsapi::Handle<$T> {
+                self.0.handle()
+            }
+        }
+    }
+}
+
+pub trait AsHandleMut<T> : AsHandle<T> {
+    fn handle_mut(&mut self) -> MutableHandle<T>;
+
+    fn set(&mut self, v: T) where T: Copy {
+        self.handle_mut().set(v)
+    }
+}
+
+/// Given the types `$T` and `$U`, implements the trait `AsHandleMut<$T>` for the type `$U`, where
+/// `$U` is a newtype whose underlying type implements `HandleMut<$T>`.
+#[macro_export]
+macro_rules! derive_as_handle_mut {
```

Comment:
I don't really see a purpose for defining these macros here. The implementations aren't hard to write by hand, and I haven't come up with a use case for this in Servo yet.